### ISSUE TITLE
Stack references for functions using rsp to track stack frame

### ISSUE THREAD

**Issue by HQ1995** [NationalSecurityAgency/ghidra#911]
I noticed that if a function only uses rsp to track stack frame (the rbp register is used as a regular register), stack references and offsets are incorrect.

How can I get correct offsets? 
Is there any rsp based stack analysis in ghidra?

**Comment by Nalen98** [comment]
I have the same question (decompile eBPF Processor for Ghidra)!!! eBPF uses only frameponter register, it’s unchangeable, and we use this register and offset to access the stack (such as index in array)
Check out my issue...
https://github.com/Nalen98/eBPF-for-Ghidra/issues/2

**Comment by emteere** [comment]
The stack analysis in Ghidra is only RSP based.  All stack variables are relative to RSP at entry to the function.  The function can use any register for the frame, or none at all.  If the RSP is moved into any register and a reference is seen to the stack, a stack reference is created relative to RSP.  This is different than other disassemblers or decompilers that use offsets from the EBP register after the frame is established.

There have been thoughts of providing an option to show references/offsets relative to some a frame register as an option to show the same offsets as say IDA or a debugger that may rely on an established frame to show values.

We can take a look, at the example to see whats going on.  It is also possible that an instruction with incorrect semantics is being used and needs to be fixed, or a call to a function is messing with the stack in a hidden way.

That said, it doesn't mean there isn't some issue...

**Comment by emteere** [comment]
@HQ1995, can you provide an example set of bytes for a function that has this issue?

**Comment by HQ1995** [comment]
@emteere, In my case, there is a function directly uses rsp register to reference the stack variable. 

At first, ghidra analyzed the stack(newStackAnalysis) automatically, the variable in offset 0x78 has no appeared. 
![image](https://user-images.githubusercontent.com/7395848/66285728-081a6300-e89c-11e9-9ac2-ac85a89b5a49.png)


Then I manually committed local variables (right-click the function name) for this function and redo the stack analysis. 

Although the stack array appeared, the reference has not appeared yet. 
![image](https://user-images.githubusercontent.com/7395848/66285708-f042df00-e89b-11e9-9aa6-a37656e6bedd.png)

Thanks for your help!

**Comment by emteere** [comment]
Hadn't seen this had been updated, until I went back through.

The stack references created on the listing are based purely on a read/write access to the stack.  In the case above a pointer on the stack was put into a parameter that was then passed to a function.  We currently don't detect this access.
The code does look at parameters to functions for constants passed that are references.  It just doesn't consider stack offsets passed in parameters.  So it should be easy to implement.
